### PR TITLE
Use hardcoded fallback when MDM Server mDNS resolution fails

### PR DESCRIPTION
**Pull request by pentestiing**
Jira-Id: SECO-5072

### CODE REVIEW

**Review by TIISR — CHANGES_REQUESTED** [review]
Lol sorry I say "fdbb" for short but it should be the same 64bit prefix fdbb:... and only then ::1 🙂

File: modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py [hunk 1]
```diff
@@ -66,6 +66,7 @@ def __init__(
         self.__debug_config_interval: int = Constants.FAIL_POLLING_TIME_SECONDS.value  # TODO: possible to update from MDM server?
 
         self.__url: str = "defaultmdm.local:5000"  # mDNS callback updates this one
+        self.__fallback_url: str = "[fdbb::1]:5000"
```
Comment (saauvine) [inline modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py:69]: Not sure did I understand Seb's comment correctly but should this be then:
self.__fallback_url: str = f"[{self.__IPV6_WHITE_PREFIX}::1]:5000"

Comment (pentestiing) [reply modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py:69]: I think so, will clarify now :+1:

**Review by TIISR — APPROVED** [review]

File: modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py
Comment (saauvine) [inline modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py:1]: I didn't remember that __IPV6_WHITE_PREFIX was defined on CBMA adaptation side thus now getting warning on MDM agent side about accessing protected member. Perhaps we should move those constant definitions later into constants.py module.

**Review by saauvine — APPROVED** [review]

File: modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py [hunk 2]
```diff
@@ -80,6 +79,8 @@ def __init__(
             self.logger,
             self.__lock
         )
+        self.__url: str = "defaultmdm.local:5000"  # mDNS callback updates this one
+        self.__fallback_url: str = f"[{self.cbma_ctrl.__IPV6_WHITE_PREFIX}::1]:5000"
```
Comment (joenpera) [inline modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py:83]: This line won't work.

**Review by joenpera — CHANGES_REQUESTED** [review]

File: modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py
Comment (pentestiing) [reply modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py:1]: Is the warning still there if they get moved to class variables?

```python
class CBMAAdaptation(object):
    """
    CBMA Control
    """
    __IPV6_WHITE_PREFIX: str = "fdbb:1ef7:9d6f:e05d"
    __IPV6_RED_PREFIX: str = "fdd8:84dc:fe30:7bf2"
...
```
:snake:

File: modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py (hunk 2 above)
Comment (pentestiing) [reply modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py:83]: Can you try this? https://github.com/tiiuae/mesh_com/pull/450#discussion_r1587258156

Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py:83]: it's still private.

Comment (saauvine) [reply modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py:83]: It should be self.__fallback_url: str = f"[{self.cbma_ctrl._CBMAAdaptation__IPV6_WHITE_PREFIX}::1]:5000" in order to work. But better way would be probably move those constant definitions to constants.py instead. Perhaps also some of the other magic prefixes from 99_at_boot could be given some purpose via naming some of those as well.

Comment (pentestiing) [reply modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py]: Fixed in https://github.com/tiiuae/mesh_com/pull/450/commits/f09732dc32dcdab1e314eaee41e5af4f41e5f59f :1st_place_medal:

File: modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py
Comment (pentestiing) [reply modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py:1]: Moved to Constants in https://github.com/tiiuae/mesh_com/pull/450/commits/f09732dc32dcdab1e314eaee41e5af4f41e5f59f :ladder:

**Review by saauvine — APPROVED** [review]

**Review by joenpera — APPROVED** [review]

**Pull request merged**